src/theory/quantifiers/quant_util.cpp

```diff
@@ -238,6 +238,23 @@ Node QuantArith::solveEqualityFor( Node lit, Node v ) {
   return Node::null();
 }
 
+bool QuantArith::decompose(Node n, Node v, Node& coeff, Node& rem) {
+  std::map<Node, Node> msum;
```

Comment:
Use unordered_map here (and as a consequence, in getMonomialSum).

Reply:
This needs to be ordered everywhere. The order corresponds to order of variables in monomial sums in the arithmetic rewriter.

Reply:
Can you maybe document this (in getMonomialSum perhaps?) so that it doesn't get changed by mistake in the future?